app/views/unattended/provisioning_templates/snippet/subscription_manager_setup.erb

```diff
@@ -88,11 +88,11 @@ fi
      end
    elsif @subman_setup_scenario == 'provisioning'
      if plugin_present?('katello')
-       server_hostname = @host.content_source.rhsm_url.host
+       server_hostname = @host.content_source.registration_url.host
```

Comment:
Using `registration_url.host` here ensures that it uses the hostname of the load balancer, not of the underlying smart proxy.